api/src/controller/account.controller.ts

```diff
@@ -68,9 +69,9 @@ export class AccountController implements interfaces.Controller {
   @httpPost('/list')
   public async list(req: express.Request, res: express.Response, next: express.NextFunction): Promise<void> {
     try {
-      this.checkControllerActive();
-      const account: Account | null = req.body.account;
-      this.roleService.checkPermission(account, ROLE_TYPES.ADMIN);
+      checkControllerActive();
+      const auth: Account | ApiKey | null = req.body.auth;
+      this.roleService.checkPermission(auth, ROLE_TYPES.ADMIN);
```

Comment:
```
const auth: Account | ApiKey | null = req.body.auth;
this.roleService.checkPermission(auth, ROLE_TYPES.XXX);
```

seems repetitive, would a middleware be helpful?

Reply:
It is indeed repetitive. The only problem is that the ROLE_TYPES.XXX is based on the specific controller endpoint. I will see if it's possible to use a local middleware on the controller endpoint instead of a universal middleware.